From: Ondrej Jirman <megous@megous.com>
Date: Fri, 28 Jan 2022 17:35:08 +0100
Subject: input: kb151: Return EXDEV when MCU tells us that the charger is
 sleeping

This helps the IP5XXXX driver do a saner reporting of the charging status.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/input/keyboard/kb151.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/keyboard/kb151.c b/drivers/input/keyboard/kb151.c
index 453eee2..e0c808b 100644
--- a/drivers/input/keyboard/kb151.c
+++ b/drivers/input/keyboard/kb151.c
@@ -362,7 +362,7 @@ static int kb151_charger_regmap_read(void *context,
 			dev_err(&client->dev,
 				"Charger read failed - MCU returned 0x%hhx\n",
 				buf3[1]);
-			return -EIO;
+			return -EXDEV;
 		}
 	}
 
@@ -415,7 +415,7 @@ static int kb151_charger_regmap_write(void *context, unsigned int reg, unsigned
 			dev_err(&client->dev,
 				"Charger write failed - MCU returned 0x%hhx\n",
 				buf2[1]);
-			return -EIO;
+			return -EXDEV;
 		}
 	}
 
